quinn-proto/src/connection/mod.rs

```diff
@@ -111,6 +111,10 @@ where
 
     /// cid length used to decode short packet
     local_cid_len: usize,
+    /// The "real" local IP address which was was used to receive the initial packet.
+    /// This is only populated for the server case, and if known
+    initial_local_ip: Option<IpAddr>,
```

Comment:
I'm not sure about `initial` here. Servers can't migrate, and if they did, we'd want to update the source address we send from. Maybe just `local_ip`, and note in the comment that it's the source address for outgoing UDP packets?

Reply:
> Servers can't migrate
I don't actually remember what the spec says about this, since it's a bit of a weird use-case.

I called it `local_` since it's currently only updated on creation, and not on every packet. Doing the latter would at least be wrong for the use-case I have, where outgoing packets need to be tagged with this target IP. It would seem to allow the client to move the server to use different paths, if it can "guess" other server IPs. 

That said I also wasn't fully sure about the name - I just couldn't come up with a better one. Initially I just had `destination_ip`, but that is even more confusing since it it's not the address the server is sending to but it's own one.

Reply:
I don't think we'd ever want to update it from incoming packets regardless; since server migration isn't supported it won't change at all at present, and if it became supported, we'd want to update the address explicitly when we migrate.

Reply:
So you would just having `fn local_ip()`? And maybe document that this is only populated for servers when the connection is established? I can live with that too.

Reply:
Yeah, sounds sensible.